pkg/common/resources/resources_test.go

```diff
@@ -1205,6 +1205,110 @@ func TestGetShares(t *testing.T) {
 	}
 }
 
+func TestCompUsageRatio(t *testing.T) {
+	tests := []struct {
+		left     *Resource
+		right    *Resource
+		total    *Resource
+		expected int
+		message  string
+	}{
+		{
+			left:     nil,
+			right:    nil,
+			total:    nil,
+			expected: 0,
+			message:  "nil resources",
+		},
+		{
+			left:     NewResource(),
+			right:    NewResource(),
+			total:    nil,
+			expected: 0,
+			message:  "empty resource with total nil",
+		},
+		{
+			left:     NewResource(),
+			right:    NewResource(),
+			total:    NewResource(),
+			expected: 0,
+			message:  "empty resources",
+		},
+		{
+			left:     &Resource{Resources: map[string]Quantity{"zero": 0}},
+			right:    &Resource{Resources: map[string]Quantity{"zero": 0}},
```

Comment:
Is there a particular reason why you directly create the type instead of calling `NewResourceXXX()`?

Reply:
I refer other test case use this way.
Thanks for the suggestion, I will change to use method call.